lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3888,6 +3922,26 @@ SPIRVValue *LLVMToSPIRVBase::transIntrinsicInst(IntrinsicInst *II,
     return BM->addExtInst(STy, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp, Ops,
                           BB);
   }
+  case Intrinsic::frexp: {
+    if (!checkTypeForSPIRVExtendedInstLowering(II, BM))
+      break;
+    SPIRVWord ExtOp = getBuiltinIdForIntrinsic(IID);
+
+    SPIRVType *FTy = transType(II->getType()->getStructElementType(0));
+    SPIRVTypePointer *ITy = static_cast<SPIRVTypePointer *>(transPointerType(
+        II->getType()->getStructElementType(1), SPIRAS_Private));
+
+    assert(ITy->getElementType()->getBitWidth() == 32);
```

Comment:
Is this check in conformance with comments? Do we want to assert out if type is i16?
Also two other comments:
1. Can we pls add a test for this?
2. Can we add a error message here?

Thanks